x-pack/legacy/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -316,6 +342,12 @@ async function getLiveElasticsearchCollectionEnabled(req) {
 export const getCollectionStatus = async (req, indexPatterns, clusterUuid, nodeUuid, skipLiveData) => {
   const config = req.server.config();
   const kibanaUuid = config.get('server.uuid');
+  const hasPermissions = await hasNecessaryPermissions(req);
+  if (!hasPermissions) {
+    return {
+      hasPermissions: false
+    };
+  }
```

Comment:
I think we should also include `hasPermissions: true` somehow into `return status;` bellow, or else it won't work with a falsey check eg: `if (!hasPermissions)` if you try it here: https://github.com/elastic/kibana/pull/50421/files#diff-b8005a1665fdee32810ccba3af240ed3R85

Reply:
Good point, but my check is `data.hasPermissions === false`, not `!hasPermissions`? I think this works fine, at least in my testing

Reply:
It does indeed, but my point is what if one of us decide to expand on this logic later on in the future and use `!hasPermissions` which won't work (since we'll most likely forget that it's an explicit `false`). I think it should work both ways tbh

Reply:
Sure, fixed up in https://github.com/elastic/kibana/pull/50421/commits/4970da2e72b2446bb4039dcbe1f37168a48bc1d1